### PR TITLE
add memory_storage using ann to avoid similar idea repetitive execution

### PR DESCRIPTION
**Pull request by better629**
1. Use FaissStore as MemoryStorage to store the watched action's related messages of a role.
2. Startup with similar idea will return with `no news` in log directly due to the existed memory.

### CODE REVIEW

File: metagpt/roles/role.py [hunk 1]
```diff
@@ -174,13 +184,7 @@ async def _observe(self) -> int:
...
         observed = self._rc.env.memory.get_by_actions(self._rc.watch)
         
-        already_observed = self._rc.memory.get()
-        
-        news: list[Message] = []
-        for i in observed:
-            if i in already_observed:
-                continue
-            news.append(i)
+        news = self._rc.memory.remember(observed)  # remember recent exact or similar memories
```
Comment (geekan) [inline metagpt/roles/role.py:187]: remember一般会同时存在长期和短期记忆，短期记忆比如最近10个

Comment (better629) [reply metagpt/roles/role.py:187]: > remember一般会同时存在长期和短期记忆，短期记忆比如最近10个

@geekan Do you means that remember from short-term mem first and then integrate with the long-term result ?

Comment (better629) [reply metagpt/roles/role.py:187]: @geekan integrate stm & ltm with latest commit

**Review by shenchucheng — APPROVED** [review]
LGTM

**Comment by geekan** [comment]
LGTM

**Pull request merged**